src/libraries/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -447,14 +459,26 @@ public override Task WriteAsync(byte[] array, int offset, int count, Cancellatio
                 return Task.CompletedTask; // return immediately if no bytes to write; no need for overhead.
 
             Memory<byte> buffer = new Memory<byte>(array, offset, count);
-            SerialStreamIORequest result = new SerialStreamIORequest(cancellationToken, buffer);
+            SerialStreamIORequest result = new(cancellationToken, buffer);
             _writeQueue.Enqueue(result);
 
             EnsureIOLoopRunning();
 
             return result.Task;
         }
 
+#if NETCOREAPP
+        public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken = default)
+        {
+            SerialStreamIORequest result = new(cancellationToken, buffer);
+            _writeQueue.Enqueue(result);
+
+            EnsureIOLoopRunning();
+
+            return new ValueTask(result.Task);
+        }
+#endif
```

Comment:
It would be nice to reduce the code duplication and the number of `#if` by doing something similar to:

```cs
        public override Task<int> ReadAsync(byte[] array, int offset, int count, CancellationToken cancellationToken)
        {
            CheckReadWriteArguments(array, offset, count);

            return count == 0 
                ? Task<int>.FromResult(0) // return immediately if no bytes requested; no need for overhead.
                : ReadInternalAsync(new Memory<byte>(array, offset, count), cancellationToken);
        }

        private Task<int> ReadInternalAsync(Memory<byte> buffer, CancellationToken cancellationToken)
        {
            SerialStreamIORequest result = new SerialStreamIORequest(cancellationToken, buffer);
            _readQueue.Enqueue(result);

            EnsureIOLoopRunning();

            return result.Task;
        }

        public override Task WriteAsync(byte[] array, int offset, int count, CancellationToken cancellationToken)
        {
            CheckWriteArguments(array, offset, count);

            return count == 0
                ? Task.CompletedTask // return immediately if no bytes to write; no need for overhead.
                : WriteInternalAsync(new ReadOnlyMemory<byte>(array, offset, count), cancellationToken);
        }

        private Task WriteInternalAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken)
        {
            SerialStreamIORequest result = new SerialStreamIORequest(cancellationToken, buffer);
            _writeQueue.Enqueue(result);

            EnsureIOLoopRunning();

            return result.Task;
        }

#if NETCOREAPP
        public override ValueTask<int> ReadAsync(Memory<byte> buffer, CancellationToken cancellationToken = default)
            => new ValueTask<int>(ReadInternalAsync(buffer, cancellationToken));

        public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken = default)
            => new ValueTask(WriteInternalAsync(buffer, cancellationToken));
#endif
```

Reply:
Fantastic feedback again @adamsitnik. Thanks 👍